ddtrace/contrib/django/patch.py

```diff
@@ -1,17 +1,513 @@
+    if hasattr(django.conf.settings, 'MIDDLEWARE') and django.conf.settings.MIDDLEWARE:
+        settings_middleware += django.conf.settings.MIDDLEWARE
+    if hasattr(django.conf.settings, 'MIDDLEWARE_CLASSES') and django.conf.settings.MIDDLEWARE_CLASSES:
+        settings_middleware += django.conf.settings.MIDDLEWARE_CLASSES
+
+    for mw_path in settings_middleware:
+        split = mw_path.split('.')
+        if len(split) > 1:
+            mod = '.'.join(split[:-1])
+            attr = split[-1]
+            mw = django.utils.module_loading.import_string(mw_path)
+            if isfunction(mw) and not isinstance(mw, wrapt.ObjectProxy):
+                # Function-based middleware is a factory which returns the function used to handle the requests.
+                # So instead of wrapping the factory, we want to wrap its returned value.
+                def traced_factory(func, instance, args, kwargs):
+                    # r is the middleware handler function returned from the factory
+                    r = func(*args, **kwargs)
+                    return FunctionWrapper(r, traced_func(django, 'django.middleware', resource=mw_path))
+                wrap(mod, attr, traced_factory)
+            elif isclass(mw):
+                for hook in ['process_request', 'process_response', 'process_view',
+                             'process_exception', 'process_template_response', '__call__']:
+                    if hasattr(mw, hook) and not isinstance(getattr(mw, hook), wrapt.ObjectProxy):
+                        wrap(mod, '{0}.{1}'.format(attr, hook),
+                             traced_func(django, 'django.middleware', resource=mw_path + '.{0}'.format(hook)))
+    return func(*args, **kwargs)
+
+
+@with_traced_module
+def traced_get_response(django, pin, func, instance, args, kwargs):
+    """Trace django.core.handlers.base.BaseHandler.get_response() (or other implementations).
+
+    This is the main entry point for requests.
+
+    Django requests are handled by a Handler.get_response method (inherited from base.BaseHandler).
+    This method invokes the middleware chain and returns the response generated by the chain.
     """
+
+    request = args[0] if len(args) > 0 else kwargs.get('request', None)
+    if request is None:
+        return func(*args, **kwargs)
+
+    try:
+        # Django 2.2.0 added `request.headers` which is a case sensitive dict of headers
+        # For previous versions use `request.META` a dict with `HTTP_X_UPPER_CASE_HEADER` keys
+        if django.VERSION >= (2, 2, 0):
+            request_headers = request.headers
+        else:
+            request_headers = request.META
+
+        if config.django['distributed_tracing_enabled']:
+            context = propagator.extract(request_headers)
+            if context.trace_id:
+                pin.tracer.context_provider.activate(context)
+
+        # Determine the resolver and resource name for this request
+        if hasattr(request, 'urlconf'):
```

Comment:
I wonder if we can just do `resolver = get_resolver(getattr(request, 'urlconf', None))`, what happens if we do `get_resolver(None)`?